### ISSUE TITLE
LTS Contract

### ISSUE THREAD

**Issue by kilrau** [raiden-network/raiden#4629]
## Abstract

Settle on a stable, LTS contract to be supported by all raiden releases and `develop` branch for 6 months+.

## Motivation

In order to use the latest raiden release or `develop`, one has to use the latest contracts. Setups, like https://github.com/ExchangeUnion/xud-simnet/ can't just re-deploy contracts, hence a LTS contract is desirable.

## Specification

a) Required contract version will not change in new Raiden releases or optionally an additional "production" mode which allows to use this LTS contract.
b) be able to open large channels (no limits in the LTS contract, nor client)
c) be able to register multiple tokens on the LTS contract

I think that b) and c) are already implemented in the latest contracts.

## Backwards Compatibility

See above.

**Comment by LefterisJP** [comment]
Hey @kilrau the last LTS contract was Red Eyes which has been live for more than 6 months. The next one will be the Alderaan release. Is that satisfactory?

**Comment by kilrau** [comment]
Problem is that the Red Eyes 0.4.0 contract is currently *not* supported by the latest raiden release as per @hackaugusto . If it was and if Alderaan will, that would be satisfactory, yes :)

Since we currently don't have a stable ("not-to-be-changed") contract for the latest raiden release, I would define the goal of this issue to settle on a stable Alderaan contract as soon as possible. Once that's done, we can close here.

**Comment by LefterisJP** [comment]
At the point the Red Eyes contracts were released and the Red Eyes release was made, that release is supported by us if there is any bugs or anything. But current develop are not-compatible with Red Eyes, since it's a different branch entirely.

Once we release Alderaan something similar will happen.

**Comment by kilrau** [comment]
That's quite a problem for us, since we can only use these infrequent mainnet releases then, even though on our end we want to run raiden on our own private chain (SimNet) and not mainnet. It is highly desirable for us to use the latest raiden version on our Simnet and also Testnet, since this version fixes a lot of things that we can't expect to be backported to Red Eyes (e.g. compatibility with latest geth which is the only version with fast sync working properly). Ergo, it would be very nice to have the Alderaan contract be supported by later releases too.

**Comment by LefterisJP** [comment]
> Ergo, it would be very nice to have the Alderaan contract be supported by later releases too.

Oh it would be. I did not mean to say it would not. What I meant is that there would probably either be 2 branches or more likely 2 modes, development and production, determined by CLI flags which would either get you in the stable LTS Alderaan version or the experimental development contracts mode.

**Comment by kilrau** [comment]
That sounds great!

**Comment by rakanalh** [comment]
@kilrau Is the answer from Lefteris sufficient for closing the ticket?

**Comment by kilrau** [comment]
The answer yes. But none of the described things are done yet as far as I'm aware.

I think we should leave this open until
- [ ] the Alderaan contract is fixed
- [ ] there is a CLI flag to use the Alderaan production contract throughout the next releases

**Comment by pirapira** [comment]
> there is a CLI flag to use the Alderaan production contract throughout the next releases

We haven't chosen an approach here. Maybe there will be a separate series of releases that talks with Alderaan contracts.

> the Alderaan contract is fixed

When it's deployed, I'll check this box.

**Comment by kilrau** [comment]
Just to capture what we discussed on gitter: It might be a good way to let the raiden client connect to [proxy contracts](https://fravoll.github.io/solidity-patterns/proxy_delegate.html) to be able to upgrade the actual contracts in a non-breaking way in future.

**Comment by pirapira** [comment]
@kilrau thanks for the idea.  I haven't thought about it.  I don't think we're going to have the delegation pattern soon.  One reason is to avoid the onchain complexity, and the other reason is to allow channel participants to be certain about whatever can happen onchain.

The situation about the onchain complexity is like this.  So far we've been trying to keep the onchain components as simple as possible, and we've tolerated some offchain complexities for that.  This is slowly changing (we started putting more information in events though it can be looked up in the storage, etc).

I would see an additional attack vector in the update mechanism.  Once the private key of the controller is stolen, all funds on the contract will disappear.
